protocol/app/prepare/prepare_proposal.go

```diff
@@ -71,8 +74,21 @@ func PrepareProposalHandler(
 			return &EmptyResponse, nil
 		}
 
+		var extCommitBzTx []byte
+		if len(req.Txs) >= slinkyabci.NumInjectedTxs {
+			extCommitBzTx = req.Txs[slinkyabci.OracleInfoIndex]
+		}
+
+		// get the update market prices tx
+		msg, err := priceUpdateGenerator.GetValidMarketPriceUpdates(ctx, extCommitBzTx)
```

Comment:
Maybe I'm missing something but this doesn't exclude the markets whose prices changed less than a threshold, right?

Reply:
My understanding is that this is handled in the Prices Keeper. 
https://github.com/dydxprotocol/v4-chain/pull/1139/files#diff-0da397be90196d47fd5225c12b6bfd6c7c4adc27b46e334f33776f510c0c1031R54

We only extend vote with "valid" market price updates. If a price update is not above the min_price threshold it would not be returned by that keeper method. Perhaps I'm mistaken though and this needs to be handled/validated outside the keeper?

Reply:
makes sense!